clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -148,6 +150,33 @@ public boolean matches(ClientRequest request) {
     }
 
     @Test
+    public void testFetchedRecordsRaisesOnSerializationErrors() {
+        ByteArrayDeserializer deserializer = new ByteArrayDeserializer() {
+            @Override
+            public byte[] deserialize(String topic, byte[] data) {
+                throw new SerializationException();
```

Comment:
Well, we wouldn't want to just remove that clause -- in the case of `SerializationException` you want to maintain the `SerializationException`, but if there's some other `RuntimeException` (which there can easily be for serializers that aren't aggressively catching exceptions and converting to `SerializationException`) then you still need to convert it to a basic `KafkaException`.

I think you could do this:

```
try {
  // parse record
} catch (SerializationException e) {
  throw new SerializationExceptionException("Error deserializing key/value for partition " + partition + " at offset " + logEntry.offset(), e);
} catch (RuntimeException e) {
  throw new KafkaException("Error deserializing key/value for partition " + partition + " at offset " + logEntry.offset(), e);
}
```

as long as we're confident there aren't any other `KafkaExceptions` we'd want to handle differently (i.e. any other more specific types of `KafkaException` where we'd want to preserve the same type instead of generalizing to `KafkaException`).